app/code/Magento/Cms/Block/BlockByIdentifier.php

```diff
@@ -0,0 +1,170 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Cms\Block;
+
+use Magento\Cms\Api\Data\BlockInterface;
+use Magento\Cms\Api\GetBlockByIdentifierInterface;
+use Magento\Cms\Model\Block as BlockModel;
+use Magento\Cms\Model\Template\FilterProvider;
+use Magento\Framework\DataObject\IdentityInterface;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Framework\View\Element\AbstractBlock;
+use Magento\Framework\View\Element\Context;
+use Magento\Store\Model\StoreManagerInterface;
+
+/**
+ * This class is replacement of \Magento\Cms\Block\Block, that accepts only `string` identifier of CMS Block
+ */
+class BlockByIdentifier extends AbstractBlock implements IdentityInterface
+{
+    public const CACHE_KEY_PREFIX = 'CMS_BLOCK';
+
+    /**
+     * @var GetBlockByIdentifierInterface
+     */
+    private $blockByIdentifier;
+
+    /**
+     * @var StoreManagerInterface
+     */
+    private $storeManager;
+
+    /**
+     * @var FilterProvider
+     */
+    private $filterProvider;
+
+    /**
+     * @var BlockInterface
+     */
+    private $cmsBlock;
+
+    /**
+     * @param GetBlockByIdentifierInterface $blockByIdentifier
+     * @param StoreManagerInterface $storeManager
+     * @param FilterProvider $filterProvider
+     * @param Context $context
+     * @param array $data
+     */
+    public function __construct(
+        GetBlockByIdentifierInterface $blockByIdentifier,
+        StoreManagerInterface $storeManager,
+        FilterProvider $filterProvider,
+        Context $context,
+        array $data = []
+    ) {
+        parent::__construct($context, $data);
+        $this->blockByIdentifier = $blockByIdentifier;
+        $this->storeManager = $storeManager;
+        $this->filterProvider = $filterProvider;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    protected function _toHtml(): string
+    {
+        try {
+            return $this->filterOutput(
+                $this->getCmsBlock()->getContent()
+            );
+        } catch (NoSuchEntityException $e) {
```

Comment:
If there is a block in the layout using the `Magento\Cms\Block\BlockByIdentifier` class with an invalid / unknown identifier it'd probably raise some eyebrows if this exception isn't logged somewhere.

It will most likely get trampled on by all the other logs but at the same time it might be a valuable clue for someone in the future.

Reply:
That was also actually something I was thinking about, as some of Administrators would like to have the block displayed conditionally - eg. for specific store, but don't have an error when the block by identifier is missing.

Also - logging a warning / notice for every single display of a page does not sound like good approach.

Reply:
I agree, it's a good point and didn't think about it that way 👍